// @okta/okta-angular/esm2015/src/okta/components/login-redirect.component.js
/*
 * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */
import * as tslib_1 from "tslib";
import { Component } from '@angular/core';
import { OktaAuthService } from '../services/okta.service';
import * as ɵngcc0 from '@angular/core';
let OktaLoginRedirectComponent = class OktaLoginRedirectComponent {
    constructor(okta) {
        this.okta = okta;
    }
    ngOnInit() {
        const originalUri = this.okta.getOriginalUri();
        if (!originalUri) {
            this.okta.setOriginalUri('/');
        }
        this.okta.signInWithRedirect();
    }
};
OktaLoginRedirectComponent.ɵfac = function OktaLoginRedirectComponent_Factory(t) { return new (t || OktaLoginRedirectComponent)(ɵngcc0.ɵɵdirectiveInject(OktaAuthService)); };
OktaLoginRedirectComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: OktaLoginRedirectComponent, selectors: [["ng-component"]], decls: 0, vars: 0, template: function OktaLoginRedirectComponent_Template(rf, ctx) { }, encapsulation: 2 });
OktaLoginRedirectComponent = tslib_1.__decorate([ tslib_1.__metadata("design:paramtypes", [OktaAuthService])
], OktaLoginRedirectComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OktaLoginRedirectComponent, [{
        type: Component,
        args: [{ template: `` }]
    }], function () { return [{ type: OktaAuthService }]; }, null); })();
export { OktaLoginRedirectComponent };
